test/integration/pdf_generation_test.exs

```diff
@@ -131,6 +137,53 @@ defmodule ChromicPDF.PDFGenerationTest do
         path -> refute File.exists?(path)
       end
     end
+
+    # The wait_for tests intentionally use umlauts in selectors and attributes.
+    @tag :pdftotext
+    test "it waits until defined selectors have given attribute when printing from `:url`" do
+      params = [
+        wait_for: %{selector: "#print-readyö", attribute: "ready-to-printö"}
+      ]
+
+      print_to_pdf({:url, "file://#{@test_dynamic_html}"}, params, fn text ->
+        assert String.contains?(text, "Dynamic content from Javascript")
+      end)
+    end
+
+    @tag :pdftotext
+    test "it waits until defined selectors have given attribute when printing from `:url` with ISO-8859 (latin1) encoding" do
+      params = [
+        wait_for: %{selector: "#print-readyö", attribute: "ready-to-printö"}
+      ]
+
+      print_to_pdf({:url, "file://#{@test_dynamic_latin1_html}"}, params, fn text ->
+        assert String.contains?(text, "Dynamic content from Javascript")
+      end)
+    end
+
+    @tag :pdftotext
+    test "it waits until defined selectors have given attribute when printing from `:html`" do
+      params = [
+        wait_for: %{selector: "#print-readyö", attribute: "ready-to-printö"}
+      ]
+
+      print_to_pdf({:html, File.read!(@test_dynamic_html)}, params, fn text ->
+        assert String.contains?(text, "Dynamic content from Javascript")
+      end)
+    end
+
+    test "it raises RuntimeError if defined selector already has the given attribute" do
+      params = [
+        wait_for: %{selector: "#print-readyö", attribute: "ready-to-printö"}
+      ]
+
+      assert_raise(
+        RuntimeError,
+        fn ->
+          ChromicPDF.print_to_pdf({:url, "file://#{@test_dynamic_pre_existing_html}"}, params)
+        end
+      )
+    end
```

Comment:
For the others (`test_dynamic_latin1_html` and `test_dynamic_pre_existing_html`), could you please remove them again? 😄  Thanks for writing them and finding out about the ISO 8859 charset, but in the end I prefer not having to maintain them over the additional test coverage. Especially since neither of them really covers any functionality (= code) in ChromicPDF: One tests a feature of the HTML spec, the other actually tests a limitation in the library.

Again, hope you don't mind the pickyness.

Reply:
No worries regarding the pickyness, appreciate your diligence. It's more or less expected when hopping into unfamiliar code without initial discussions with the maintainers.

I'll remove these additions, no need to take the burden with them. I should have time this evening to make the changes.